src/client/clang_command.cc

```diff
@@ -120,7 +120,12 @@ bool ClangCommand::FillFlags(base::proto::Flags* flags,
       replaced_command = std::regex_replace(
           replaced_command, version_regex, "/lib/clang/" + clang_major_version);
 
-      const String self_path = base::GetSelfPath();
+      String self_path;
+      String error;
+      if (!base::GetSelfPath(&self_path, &error)) {
+        LOG(WARNING) << "Failed to get executable path: " << error;
```

Comment:
In the almost-impossible case when we fail to `GetSelfPath` old version returns empty string, which gets improperly substituted for `clang_path` component later. This version avoids such behaviour and signals to the caller that we were unable to fill flags. This does not seem to be a fatal error, though: the code in `clang.cc` simply executes the command "as is" thereafter.